Hazelcast.Net/Hazelcast.Client.Spi/ClientInvocationService.cs

```diff
@@ -269,10 +269,17 @@ private void HandleInvocationException(ClientInvocation invocation, Exception ex
                     }
                 }
                 //Fail with exception
-                if (!invocation.Future.IsComplete)
+                try
                 {
                     invocation.Future.Exception = exception;
                 }
+                catch (InvalidOperationException e)
+                {
+                    if (Logger.IsFinestEnabled())
```

Comment:
We made this log as warning at Java: https://github.com/hazelcast/hazelcast/blob/master/hazelcast/src/main/java/com/hazelcast/spi/impl/AbstractInvocationFuture.java#L377  This may be also set as warning.